src/streamlink/stream/hls.py

```diff
@@ -299,6 +300,13 @@ def iter_segments(self):
         total_duration = 0
         while not self.closed:
             for sequence in filter(self.valid_sequence, self.playlist_sequences):
+                if self.init_map != sequence.segment.map:
```

Comment:
I think a bit of a comment describing the mechanism might be useful here.

Reply:
The map segment has to be put in the queue only once, and is then valid for all following segments until an `ext_x_discontinuity` tag arrives (in this case, a possibly new map segment follows) or the stream ends.

Reply:
I want to save the init maps in a separate list and not in each `Segment` as before, because I want only the index of the init. map store, so less memory is used, especially with very long playlist of vod’s - do you have any objections, @beardypig, @bastimeyer?